cfme/tests/openstack/cloud/test_instances.py

```diff
@@ -176,12 +176,12 @@ def test_list_vms_infra_node(appliance, provider, soft_assert):
     host_collection = appliance.collections.hosts
     # Match hypervisors by IP with count of running VMs
     hvisors = {hv.host_ip: hv.running_vms for hv in provider.mgmt.api.hypervisors.list()}
+    hv_ips = hvisors.keys()
 
     # Skip non-compute nodes
     hosts = [host for host in host_collection.all(provider) if 'Compute' in host.name]
     assert hosts
-    for host in hosts:
-        host_ip = host.get_detail('Properties', 'IP Address')
+    for counter, host in enumerate(hosts):
         vms = int(host.get_detail('Relationships', 'VMs'))
-        soft_assert(vms == hvisors[host_ip],
+        soft_assert(vms == hvisors[hvisors[hv_ips[counter]]],
```

Comment:
**QUESTION**
What's happening here?

Reply:
IP Addresses returned by provider.mgmt.api.hypervisors.list() don't match hosts IPs returned by host.get_detail('Properties', 'IP Address') and this causes a Keyerror on hvisors dictionary when using host_ip as a key. because of that I used IPs in hv_ips list as keys for accessing hvisors dictionary values and counter is list iterator returned byb enumerate

Reply:
I don't understand how does this work `hvisors[hvisors[hv_ips[counter]]]`? Here `hv_ips[counter]` you get a host_ip, here `hvisors[hv_ips[counter]]`you get a value corresponding host_ip and what does happen next? Are you trying to get a value by a value, but not by a key?

Reply:
You are right, this a mistake. looks like copy paste issue
I am committing a fix